### PR TITLE
INT-483: Add missing endpoints to n8n-onfleet node

### PR DESCRIPTION
**Pull request by YokySantiago**
### feat: added team endpoints to onfleet node

- Added team auto-dispatch and driver time estimate endpoints to Onfleet node

### CODE REVIEW

File: packages/nodes-base/nodes/Onfleet/Onfleet.node.ts [hunk 1]
```diff
@@ -555,6 +557,47 @@ export class Onfleet implements INodeType {
...
+			/* -------------------------------------------------------------------------- */
+			const dropoff = this.getNodeParameter('dropoff', item) as boolean;
+			const pickup = this.getNodeParameter('pickup', item) as boolean;
+			if (!dropoff && !pickup) throw new Error('At least 1 of dropoffLocation or pickupLocation must be selected');
+			const longitudeField = `${dropoff ? 'dropoff' : 'pickup'}Longitude`;
+			const latitudeField = `${dropoff ? 'dropoff' : 'pickup'}Latitude`;
+			const longitude = this.getNodeParameter(longitudeField, item) as string;
+			const latitude = this.getNodeParameter(latitudeField, item) as string;
+
+			const wokrerTimeEstimates = {} as OnfleetWorkerEstimates;
```
Comment (0xsemaj) [inline packages/nodes-base/nodes/Onfleet/Onfleet.node.ts]: typo: `workerTimeEstimates`

File: packages/nodes-base/nodes/Onfleet/Onfleet.node.ts [hunk 2]
```diff
@@ -555,6 +557,47 @@ export class Onfleet implements INodeType {
...
+			if (pickup) {
+				wokrerTimeEstimates.pickupLocation = `${longitude},${latitude}`;
+				wokrerTimeEstimates.pickupTime = (new Date(this.getNodeParameter('pickupTime', item) as Date)).getTime() / 1000;
+			}
+			if(dropoff) {
+				wokrerTimeEstimates.dropoffLocation = `${longitude},${latitude}`;
+			}
+
+			const additionalFields = this.getNodeParameter('additionalFields', item) as IDataObject;
+			Object.assign(wokrerTimeEstimates, additionalFields);
```
Comment (0xsemaj) [inline packages/nodes-base/nodes/Onfleet/Onfleet.node.ts]: typo

Comment (0xsemaj) [reply packages/nodes-base/nodes/Onfleet/Onfleet.node.ts]: applies to every `wokrerTimeEstimates`

File: packages/nodes-base/nodes/Onfleet/descriptions/TeamDescription.ts [hunk 3]
```diff
@@ -83,6 +93,169 @@ const enableSelfAssignmentField = {
...
+const maxTasksPerRouteField = {
+	displayName: 'Max number of tasks per route',
+	name: 'maxTasksPerRoute',
+	type: 'number',
+	default: 100,
+	typeOptions: {
+		maxValue: 200,
+		minValue: 1,
+	},
+	description: 'Total number of tasks allowed on a route.',
```
Comment (0xsemaj) [inline packages/nodes-base/nodes/Onfleet/descriptions/TeamDescription.ts]: Remove all dots (period) at the end of the descriptions as suggested by n8n

File: packages/nodes-base/nodes/Onfleet/descriptions/TeamDescription.ts [hunk 4]
```diff
@@ -83,6 +93,169 @@ const enableSelfAssignmentField = {
...
+
+const serviceTImeField = {
+	displayName: 'Service time',
+	name: 'serviceTIme',
+	type: 'number',
+	default: 2,
+	typeOptions: {
+		minValue: 0,
+	},
+	description: 'The default service time to apply in Minutes to the tasks when no task service time exists.',
```
Comment (0xsemaj) [inline packages/nodes-base/nodes/Onfleet/descriptions/TeamDescription.ts]: Remove all dots (period) at the end of the descriptions as suggested by n8n

File: packages/nodes-base/nodes/Onfleet/descriptions/TeamDescription.ts [hunk 5]
```diff
@@ -83,6 +93,169 @@ const enableSelfAssignmentField = {
...
+} as INodeProperties;
+
+const routeEndField = {
+	displayName: 'Route end',
+	name: 'routeEnd',
+	type: 'string',
+	default: '',
+	description: 'Where the route will end.',
```
Comment (0xsemaj) [inline packages/nodes-base/nodes/Onfleet/descriptions/TeamDescription.ts]: Remove all dots (period) at the end of the descriptions as suggested by n8n

File: packages/nodes-base/nodes/Onfleet/descriptions/TeamDescription.ts [hunk 6]
```diff
@@ -83,6 +93,169 @@ const enableSelfAssignmentField = {
...
+		minValue: 1,
+	},
+} as INodeProperties;
+
+const dropoffField = {
+	displayName: 'Dropoff',
+	name: 'dropoff',
+	type: 'boolean',
+	default: false,
+	description: 'Dropoff.',
```
Comment (0xsemaj) [inline packages/nodes-base/nodes/Onfleet/descriptions/TeamDescription.ts]: Remove all dots (period) at the end of the descriptions as suggested by n8n

File: packages/nodes-base/nodes/Onfleet/descriptions/TeamDescription.ts [hunk 7]
```diff
@@ -83,6 +93,169 @@ const enableSelfAssignmentField = {
...
+} as INodeProperties;
+
+const pickupField = {
+	displayName: 'Pickup',
+	name: 'pickup',
+	type: 'boolean',
+	default: false,
+	description: 'Pickup.',
```
Comment (0xsemaj) [inline packages/nodes-base/nodes/Onfleet/descriptions/TeamDescription.ts]: Remove all dots (period) at the end of the descriptions as suggested by n8n

File: packages/nodes-base/nodes/Onfleet/descriptions/TeamDescription.ts [hunk 8]
```diff
@@ -83,6 +93,169 @@ const enableSelfAssignmentField = {
...
+
+const longitudeDropoffField = {
+	displayName: 'Dropoff longitude',
+	name: 'dropoffLongitude',
+	type: 'number',
+	typeOptions: {
+		numberPrecision: 14,
+	},
+	default: '',
+	description: 'The longitude for dropoff location.',
```
Comment (0xsemaj) [inline packages/nodes-base/nodes/Onfleet/descriptions/TeamDescription.ts]: Remove all dots (period) at the end of the descriptions as suggested by n8n

File: packages/nodes-base/nodes/Onfleet/descriptions/TeamDescription.ts [hunk 9]
```diff
@@ -83,6 +93,169 @@ const enableSelfAssignmentField = {
...
+
+const latitudeDropoffField = {
+	displayName: 'Dropoff latitude',
+	name: 'dropoffLatitude',
+	type: 'number',
+	typeOptions: {
+		numberPrecision: 14,
+	},
+	default: '',
+	description: 'The latitude for dropoff location.',
```
Comment (0xsemaj) [inline packages/nodes-base/nodes/Onfleet/descriptions/TeamDescription.ts]: Remove all dots (period) at the end of the descriptions as suggested by n8n

File: packages/nodes-base/nodes/Onfleet/descriptions/TeamDescription.ts [hunk 10]
```diff
@@ -83,6 +93,169 @@ const enableSelfAssignmentField = {
...
+
+const longitudePickupField = {
+	displayName: 'Pickup longitude',
+	name: 'pickupLongitude',
+	type: 'number',
+	typeOptions: {
+		numberPrecision: 14,
+	},
+	default: '',
+	description: 'The longitude for pickup location.',
```
Comment (0xsemaj) [inline packages/nodes-base/nodes/Onfleet/descriptions/TeamDescription.ts]: Remove all dots (period) at the end of the descriptions as suggested by n8n

File: packages/nodes-base/nodes/Onfleet/descriptions/TeamDescription.ts [hunk 11]
```diff
@@ -83,6 +93,169 @@ const enableSelfAssignmentField = {
...
+
+const latitudePickupField = {
+	displayName: 'Pickup latitude',
+	name: 'pickupLatitude',
+	type: 'number',
+	typeOptions: {
+		numberPrecision: 14,
+	},
+	default: '',
+	description: 'The latitude for pickup location.',
```
Comment (0xsemaj) [inline packages/nodes-base/nodes/Onfleet/descriptions/TeamDescription.ts]: Remove all dots (period) at the end of the descriptions as suggested by n8n

File: packages/nodes-base/nodes/Onfleet/descriptions/TeamDescription.ts [hunk 12]
```diff
@@ -83,6 +93,169 @@ const enableSelfAssignmentField = {
...
+} as INodeProperties;
+
+const pickupTimeField = {
+	displayName: 'Pickup time',
+	name: 'pickupTime',
+	type: 'dateTime',
+	default: Date.now(),
+	description: 'If the request includes pickupLocation pickupTime must be present if the time is fewer than 3 hours in the future.',
```
Comment (0xsemaj) [inline packages/nodes-base/nodes/Onfleet/descriptions/TeamDescription.ts]: Remove all dots (period) at the end of the descriptions as suggested by n8n

File: packages/nodes-base/nodes/Onfleet/descriptions/TeamDescription.ts [hunk 13]
```diff
@@ -83,6 +93,169 @@ const enableSelfAssignmentField = {
...
+			name: 'Bicycle',
+			value: 'BICYCLE',
+		},
+		{
+			name: 'Truck',
+			value: 'TRUCK',
+		},
+	],
+	default: 'CAR',
+	description: 'Vehicle types to ignore in the query.',
```
Comment (0xsemaj) [inline packages/nodes-base/nodes/Onfleet/descriptions/TeamDescription.ts]: Remove all dots (period) at the end of the descriptions as suggested by n8n

File: packages/nodes-base/nodes/Onfleet/descriptions/TeamDescription.ts [hunk 14]
```diff
@@ -83,6 +93,169 @@ const enableSelfAssignmentField = {
...
+
+const serviceTimeEstimateField = {
+	displayName: 'Service time',
+	name: 'serviceTIme',
+	type: 'number',
+	default: 120,
+	typeOptions: {
+		minValue: 0,
+	},
+	description: 'The expected time a worker will take at the pickupLocation, dropoffLocation, or both (as applicable) Unit: seconds.',
```
Comment (0xsemaj) [inline packages/nodes-base/nodes/Onfleet/descriptions/TeamDescription.ts]: Remove all dots (period) at the end of the descriptions as suggested by n8n

File: packages/nodes-base/nodes/Onfleet/descriptions/TeamDescription.ts [hunk 15]
```diff
@@ -83,6 +93,169 @@ const enableSelfAssignmentField = {
...
+	],
+	default: 'CAR',
+	description: 'Vehicle types to ignore in the query.',
+} as INodeProperties;
+
+const serviceTimeEstimateField = {
+	displayName: 'Service time',
+	name: 'serviceTIme',
+	type: 'number',
+	default: 120,
```
Comment (0xsemaj) [inline packages/nodes-base/nodes/Onfleet/descriptions/TeamDescription.ts:252]: I thought the default of this service time is 2 minutes? https://docs.onfleet.com/reference#team-auto-dispatch

File: packages/nodes-base/nodes/Onfleet/descriptions/TeamDescription.ts [hunk 16]
```diff
@@ -83,6 +93,169 @@ const enableSelfAssignmentField = {
...
+
+const scheduleTimeWindowField = {
+	displayName: 'Schedule time window (JSON)',
+	name: 'scheduleTimeWindow',
+	type: 'json',
+	default: '{}',
+	description: 'This is the Driver\'s scheduled time window. Param must be an array of length 2 in unix time in seconds precision, [start, end]',
+} as INodeProperties;
+
+const serviceTImeField = {
```
Comment (0xsemaj) [inline packages/nodes-base/nodes/Onfleet/descriptions/TeamDescription.ts]: `serviceTimeField`

**Review by jamesliupenn — CHANGES_REQUESTED** [review]

File: packages/nodes-base/nodes/Onfleet/descriptions/TeamDescription.ts (hunk 15 above)
Comment (YokySantiago) [reply packages/nodes-base/nodes/Onfleet/descriptions/TeamDescription.ts:252]: For the Get driver time estimate endpoint, it's 120 https://docs.onfleet.com/reference#delivery-estimate

File: packages/nodes-base/nodes/Onfleet/descriptions/TeamDescription.ts (hunk 16 above)
Comment (YokySantiago) [reply packages/nodes-base/nodes/Onfleet/descriptions/TeamDescription.ts]: Here the default service time it's 2 since this it's the auto dispatch endpoint

**Review by jamesliupenn — APPROVED** [review]
LGTM

**Review by jamesliupenn — APPROVED** [review]
Change looks good, merging

**Pull request merged**